en/airgap.md

```diff
@@ -0,0 +1,146 @@
+that attempts to use it, will contact it via HTTPS.
+
+On the same machine, register the air-gapped Snap Store Proxy:
+
+```bash
+# You'll be prompted to provide your SSO authentication and will
+# be asked some survey questions about the intended proxy usage.
+sudo snap-proxy register --offline --channel=edge --arch=amd64
+```
+
+The result of the above is a tarball `offline-snap-store.tar.gz` that is then
+moved to the target host machine for the air-gapped Snap Store Proxy.
+
+## Database setup
+
+The target machine (the air-gapped Snap Store Proxy host) should have network
+access to a [properly configured PostgreSQL database](install.md#database).
+
+## Installation
+
+You will need the `offline-snap-store.tar.gz` bundle from the registration step
+to continue with installation.
+
+!!! NOTE:
+    The air-gapped mode is in a closed internal beta currently and installation
+    is password protected.
+
+The script below illustrates the installation process on the target air-gapped
+machine. Please note that the following variables need to be set appropriately:
+
+* `POSTGRESQL_CONN_STRING` - the connection string to a
+  [properly set up PostgreSQL database](install.md#database)
+
+* `SNAPSTORE_BETA_PASSWORD` - a closed beta password required for the
+  installation of the air-gapped Snap Store Proxy
+
+* `PROXY_ACCESS_PASSWORD` - password required for management of the air-gapped
+  Snap Store Proxy over the network
+
+```bash
+#!/bin/bash
+
+set -eu
+
+# PostgreSQL connection string to the Snap Store Proxy database.
+POSTGRESQL_CONN_STRING="${POSTGRESQL_CONN_STRING}"
+# Closed beta password required for the airgap installation.
+SNAPSTORE_BETA_PASSWORD="${SNAPSTORE_BETA_PASSWORD}"
+# Management access password for the proxy.
+PROXY_ACCESS_PASSWORD="${PROXY_ACCESS_PASSWORD}"
+
+tar xvzf offline-snap-store.tar.gz
+cd offline-snap-store
+./install.sh
+
+sudo snap-store-proxy config proxy.db.connection="$POSTGRESQL_CONN_STRING"
+
+SNAPSTORE_BETA_PASSWORD="$SNAPSTORE_BETA_PASSWORD" sudo -E snap-store-proxy enable-airgap-mode --password $PROXY_ACCESS_PASSWORD
+
+sudo snap-store-proxy status
+```
+
+## Side-loading snaps
+
+Air-gapped Snap Store Proxy operators can fetch snaps from the official Snap
+Store and import them into their air-gapped proxy. These will be the only snaps
+(and their revisions) available for installation from the air-gapped proxy.
+
+### Fetching snaps
+
+Example of fetching the `jq` snap on a **machine with internet access**:
+
+```bash
+sudo snap-store-proxy fetch-snaps jq --channel=stable --architecture=amd64
+```
+
+This produces a `tar.gz` file that has to be moved to the air-gapped proxy and
+imported there.
+
+### Importing (pushing) snaps
+
+Example of importing a `jq.tar.gz` snap bundle on the air-gapped proxy host:
+
+```bash
+sudo snap-store-proxy push-snap jq.tar.gz
+```
+
+The `jq` snap is now available for installation from this air-gapped Snap Store
+Proxy.
+
```

Comment:
Maybe add a way to test this from a machine inside the airgap? e.g. snap info jq